cdap-docs/tools/licenses/licenses.py

```diff
@@ -207,13 +256,118 @@ def master_print():
         master_libs_dict[k].pretty_print(i+1, max)    
 
 
-def process_enterprise(input_file, options):
-    return process_dependencies(ENTERPRISE)
+def test_for_application(app):
+    import subprocess
+    try: 
+        subprocess.call(["which", app])
+        print "Found executable for '%s'" % app
+        return 1
+    except: 
+        print "No executable of '%s'" % app
+        return 0
 
 
-def process_standalone(input_file, options):
-    return process_dependencies(STANDALONE)
+def process_cdap_ui(options):
+    # Read in the current master csv file and create a structure with it
+    # Read in the checked-in dependency files:
+    #   cdap-ui/bower.json
+    #   cdap-ui/package.json
+    # Create and print to standard out the list of the references
+    # Make a list of the references for which links are missing and need to be added to the master
+    # Make a new master list (not done)
+    # Return a list:
+    #   "Dependency","Version","homepage","License","License URL","type"    
+    master_libs_dict = process_master()
+    cdap_ui_dict = {}
+    missing_libs_dict = {}
     
+    import json
+    from pprint import pprint
+    
+    for type in CDAP_UI_SOURCES.keys():
+        
+        source = CDAP_UI_SOURCES[type][0]
+        json_path = os.path.join(SCRIPT_DIR_PATH, source)
+        print "Reading '%s' dependencies file:\n%s" % (type, json_path)
+        with open(json_path) as data_file:    
+            data = json.load(data_file)
+            
+        if CDAP_UI_DEPENDENCIES_KEY in data.keys():
+            for dependency in data[CDAP_UI_DEPENDENCIES_KEY]:
+                if not dependency.startswith(CDAP_UI_CASK_DEPENDENCIES):
```

Comment:
If the dependency doesn't start with the CDAP_UI_CASK_DEPENDENCIES ("cdap-"), then it gets processed by the code. Else wise, it gets skipped. It ignores all the cdap- dependencies.